scripts/generateHeadingIDs.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const GithubSlugger = require('github-slugger');
+
+function walk(dir) {
+  let results = [];
+  const list = fs.readdirSync(dir);
+  list.forEach(function(file) {
+    file = dir + '/' + file;
+    const stat = fs.statSync(file);
+    if (stat && stat.isDirectory()) {
+      /* Recurse into a subdirectory */
+      results = results.concat(walk(file));
+    } else {
+      /* Is a file */
+      results.push(file);
+    }
+  });
+  return results;
+}
+
+function stripLinks(line) {
+  return line.replace(/\[([^\]]+)\]\([^)]+\)/, (match, p1) => p1);
+}
+
+function addHeaderID(line, slugger) {
+  // check if we're a header at all
+  if (!line.startsWith('#')) {
```

Comment:
```suggestion
  if (!line.startsWith('##')) {
```

'##' may be safer because fenced code blocks may have "pseudo-headings", which may cause problems in the future. We don't use h1-level headings (`# Foo Bar`) at all.

Reply:
I already check and make sure that we don't add headings to code blocks:

```
function addHeaderIDs(lines) {
  // Sluggers should be per file
  const slugger = new GithubSlugger();
  let inCode = false;
  const results = [];
  lines.forEach(line => {
    // Ignore code blocks
    if (line.startsWith('```')) {
      inCode = !inCode;
      results.push(line);
      return;
    }
    if (inCode) {
      results.push(line);
      return;
    }

    results.push(addHeaderID(line, slugger));
  });
  return results;
}
```

Reply:
Oh, sorry again, I failed to notice that part.